api/equality/equality.go

```diff
@@ -27,3 +30,50 @@ func TaskStatusesEqualStable(a, b *api.TaskStatus) bool {
 	copyA.Timestamp, copyB.Timestamp = nil, nil
 	return reflect.DeepEqual(&copyA, &copyB)
 }
+
+// RootCAEqualStable compares RootCAs, excluding join tokens, which are randomly generated
+func RootCAEqualStable(a, b *api.RootCA) bool {
+	switch {
+	case a == nil && b == nil:
+		return true
+	case a != nil && b != nil:
+	default:
+		return false
+	}
+	if !(bytes.Equal(a.CACert, b.CACert) &&
+		subtle.ConstantTimeCompare(a.CAKey, b.CAKey) == 1 &&
+		a.CACertHash == b.CACertHash) {
+		return false
+	}
+	switch {
+	case a.RootRotation == nil && b.RootRotation == nil:
+		return true
+	case a.RootRotation != nil && b.RootRotation != nil:
+	default:
+		return false
+	}
+	return (bytes.Equal(a.RootRotation.CACert, b.RootRotation.CACert) &&
+		subtle.ConstantTimeCompare(a.RootRotation.CAKey, b.RootRotation.CAKey) == 1 &&
+		bytes.Equal(a.RootRotation.CrossSignedCACert, b.RootRotation.CrossSignedCACert))
+}
+
+// ExternalCAsEqualStable compares lists of external CAs and determines whether they are equal
+func ExternalCAsEqualStable(a, b []*api.ExternalCA) bool {
```

Comment:
That's true - the map is created from the same object.  I was writing this from the perspective of this function can be used no matter who creates the external CAs, but happy to make a change and a comment stating the assumption.